### ISSUE TITLE
SidecarConfiguration doesn't respect management.contextPath

### ISSUE THREAD

**Issue by gbtec-ext-sergeyuzhakov** [spring-cloud/spring-cloud-netflix#2859]
Version:  1.3.6.RELEASE

If one has the code for the service like this

@EnableSidecar
@SpringBootApplication
public class ServiceApplication {

    public static void main(String[] args) {
        SpringApplication.run(ServiceApplication.class, args);
    }
}

and in application.yml has
management:
  context-path: /admin

and both the service and Eureka were started
then GET request to http://localhost:8761/eureka/apps has to contain

expected result:
<statusPageUrl>http://host:port/admin/info</statusPageUrl>
<healthCheckUrl>http://host:port/admin/health</healthCheckUrl>

actual result:
<statusPageUrl>http://host:port/info</statusPageUrl>
<healthCheckUrl>http://host:port/health</healthCheckUrl>

**Comment by ryanjbaxter** [comment]
Can you try Edgware.SR3 and report if the issue still exists?

**Comment by gbtec-ext-sergeyuzhakov** [comment]
Checked with Edgware.SR3. The issue still exists.

**Comment by ryanjbaxter** [comment]
Shouldnt you be using `sidecar.health-uri`?

**Comment by gbtec-ext-sergeyuzhakov** [comment]
I do use it. It refers to nodejs instance and works fine.
The problem is that I can't access java sidecar service from Eureka UI cause the link (see above) is broken.

**Comment by ryanjbaxter** [comment]
When I create an app with these properties
```
sidecar.port=1
spring.application.name=foo
management.context-path=/admin
```
in Eureka I see
```
<homePageUrl>http://192.168.1.116:8080/</homePageUrl>
<statusPageUrl>http://192.168.1.116:8080/admin/info</statusPageUrl>
<healthCheckUrl>http://192.168.1.116:8080/admin/health</healthCheckUrl>
```

**Comment by gbtec-ext-sergeyuzhakov** [comment]
Hello Ryan
I've created a demo for you
https://github.com/gbtec-ag/spring-cloud-sidecar-issue

**Comment by n0mer** [comment]
@spencergibb page https://cloud.spring.io/spring-cloud-netflix/multi/multi__service_discovery_eureka_clients.html still references deprecated in spring-boot 2.x property `management.context-path` (should be `management.server.servlet.context-path`, according to properties-migrator ; but for webflux-only apps `servlet` seems not relevant any more, though).

**Comment by ryanjbaxter** [comment]
@n0mer this is unrelated to this issue, please open a separate issue

**Comment by gbtec-ingogriebsch** [comment]
@ryanjbaxter @spencergibb We would like to ask if some bug fix is visible on the horizon. And we would like to know if we can help in some way to understand/solve the problem. :)

**Comment by ryanjbaxter** [comment]
@gbtec-ext-sergeyuzhakov nothing concrete yet, if you would like to look into the issue, we would welcome that.

**Comment by gbtec-ingogriebsch** [comment]
@ryanjbaxter Is it possible to have a second at this issue? Maybe there is some time after the Finchley Release? ;)

**Comment by ryanjbaxter** [comment]
Closed via 8af3ae8

**Comment by n0mer** [comment]
@ryanjbaxter is 2.x backport of this feature planned?

**Comment by ryanjbaxter** [comment]
It will be merged forward into 2.0.x.  If you need it in a build snapshot I can do that today.

**Comment by gbtec-ingogriebsch** [comment]
@ryanjbaxter Will this be part of a Spring Cloud release which is usable with Spring Boot 1.5.x (probably Edgware)?

**Comment by ryanjbaxter** [comment]
yes it will be included in Edgware.SR4

**Comment by n0mer** [comment]
> It will be merged forward into 2.0.x. If you need it in a build snapshot I can do that today.

no rush, but please ping when this code will be available in either 2.0.x artefact

**Comment by ryanjbaxter** [comment]
It should be in snapshots now
